File: scripts/build_static_frontend.py
#!/usr/bin/env python3
"""Build the buyer-facing static dashboard for Vercel/static hosts."""
from __future__ import annotations

import shutil
from pathlib import Path


ROOT = Path(__file__).resolve().parents[1]
SITE_DIR = ROOT / "frontend" / "site"
OUT_DIR = ROOT / "frontend" / "dist"


def main() -> int:
    if not SITE_DIR.exists():
        raise SystemExit(f"Static site source missing: {SITE_DIR}")

    if OUT_DIR.exists():
        shutil.rmtree(OUT_DIR)
    shutil.copytree(SITE_DIR, OUT_DIR)

    # Vercel static output should resolve clean URLs and deep links.
    for route in ("demo", "ops", "pricing"):
        route_dir = OUT_DIR / route
        index_file = route_dir / "index.html"
        if not index_file.exists():
            raise SystemExit(f"Missing route index: {index_file}")

    print(f"Static frontend build: {OUT_DIR}")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: scripts/build_static_frontend.mjs
import fs from "node:fs";
import path from "node:path";
import { fileURLToPath } from "node:url";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const root = path.resolve(__dirname, "..");
const cwdSiteDir = path.join(process.cwd(), "site");
const siteDir = fs.existsSync(cwdSiteDir) ? cwdSiteDir : path.join(root, "frontend", "site");
const outDir = fs.existsSync(cwdSiteDir) ? path.join(process.cwd(), "dist") : path.join(root, "frontend", "dist");

if (!fs.existsSync(siteDir)) {
  throw new Error(`Static site source missing: ${siteDir}`);
}

fs.rmSync(outDir, { recursive: true, force: true });
fs.cpSync(siteDir, outDir, { recursive: true });

for (const route of ["demo", "ops", "pricing", "proof"]) {
  const indexFile = path.join(outDir, route, "index.html");
  if (!fs.existsSync(indexFile)) {
    throw new Error(`Missing route index: ${indexFile}`);
  }
}

console.log(`Static frontend build: ${outDir}`);
